### PR TITLE
Copy DeckID when copying/adding note

### LINKED ISSUE #6507: "Copy note" destination is parent deck, not sub-deck
###### Reproduction Steps

1. Open Ankidroid (on Google Pixel 2), opens in "Decks" view.
2. Select desired sub-deck.
3. Go to "Card Browser" - sub-deck name and number of cards shown is displayed, as well as cards.
4. Find desired card, quick tap to enter "Edit note" (should be "Edit card"?) mode.
5. Tap "three dots on right", and select "Copy note" (should be "Copy card"?).
6. Edit note (card?) content, tap the "tick mark" to save.  This returns to "Card Browser", where the sub-deck name and the number of cards shown in 3. above is still shown, not that number plus one.

###### Expected Result
New note (card?) is placed in same sub-deck from which its source note (card?) resides in.

###### Actual Result
New note (card?) is placed in the parent deck of the sub-deck from which its source note (card?) resides in.  Now have to select parent deck in Card browser, and look through 450+ cards to find new card, long tap that card, select "three dots", tap "Change deck", and this moves the note (card) to the desired deck.

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
Reported in #6507  - Copy Note does not copy to the correct deck.

## Fixes
Fixes #6507 

## Approach
Place the DeckId in the intent, and restore to this. `mCurrentDid` overrides anything set in the model in `NoteEditor.setDid`

## How Has This Been Tested?

Tested on my phone, then added a unit test

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -522,6 +523,8 @@ public void onNothingSelected(AdapterView<?> parent) {
             }
         });
 
+        mCurrentDid = intent.getLongExtra(EXTRA_DID, 0);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:526]: We have default deck and default deck conf, what is this zero?

https://github.com/ankidroid/Anki-Android/blob/4dd43aebb17097fe72fe74fd73540c04152aa033/AnkiDroid/src/main/java/com/ichi2/libanki/Consts.java#L106-L108

File: AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java [hunk 2]
```diff
@@ -207,6 +209,31 @@ public void verifyStartupAndCloseWithNoCollectionDoesNotCrash() {
...
+        Note n = super.addNoteUsingBasicModel("Test", "Note");
+        n.model().put("did", currentDid);
+        NoteEditor editor = getNoteEditorEditingExistingBasicNote("Test", "Note", FromScreen.DECK_LIST);
+
+        getCol().getConf().put("curDeck", Consts.DEFAULT_DECK_ID); // Change DID if going through default path
+        Intent copyNoteIntent = getCopyNoteIntent(editor);
+        NoteEditor newNoteEditor = super.startActivityNormallyOpenCollectionWithIntent(NoteEditor.class, copyNoteIntent);
+
+        assertThat(editor.getDeckId(), is(currentDid));
+        assertThat(copyNoteIntent.getLongExtra(NoteEditor.EXTRA_DID, 0), is(currentDid));
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java:225]: same comment here re: the zero just to connect them - may be nothing but if it is something it's in two spots

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:526]: Zero is the value which specifies "unset, look at the model:"

https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java#L1473

I'll extract to a constant

File: AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java (hunk 2 above)
Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java:225]: Will extract as well

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: ... I didn't, it made more sense to use "use the current value if the intent isn't set".

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: Yeah I saw that, thought it was clever + correct (as was the 404 pick on the test constant ;-)

**Pull request merged**